Python/pywarpx/particle_containers.py

```diff
@@ -24,10 +24,21 @@ class ParticleContainerWrapper(object):
 
     def __init__(self, species_name):
         self.name = species_name
+        self._particle_container = None
+
+    @property
+    def particle_container(self):
+        if self._particle_container is None:
+            try:
+                mypc = libwarpx.warpx.multi_particle_container()
+                self._particle_container = mypc.get_particle_container_from_name(
+                    self.name
+                )
+            except AttributeError as e:
+                msg = "This is likely caused by attempting to access a ParticleContainerWrapper before initialize_warpx has been called"
```

Comment:
Thank you for this PR! Sorry for just seeing it now, I was just on parental leave when the PR was opened.

I thought we could also add one more fallback, right in the library loader, where we assign the `libwarpx.warpx` attribute.

In essence, we could assign an initial attribute that throws when accessed with a more distinct error message ("initialize_warpx was not yet called!"), to catch all. More specific exceptions like this one are great to add and keep.

Reply:
Oh neat idea! I can open a new PR shortly!